File: DarkOrbit.Api/Endpoints/EndpointEntity.cs
using System.Collections.Generic;
using DarkOrbit.Api.Utilities.Database;

namespace DarkOrbit.Api.Endpoints
{
    public class EndpointEntity : MongoDbEntityBase
    {
        public string Name { get; set; }

        public string Description { get; set; }

        public string ServiceId { get; set; }

        public bool RequiresAuthentication { get; set; }

        public List<EndpointMethod> EndpointMethods { get; set; }
    }
}

File: DarkOrbit.WebApp/src/app/api/models/endpoint-entity.ts
/* tslint:disable */
import { EndpointMethod } from './endpoint-method';

export interface EndpointEntity {

  id?: string;

  name?: string;

  serviceId?: string;

  requiresAuthentication?: boolean;

  endpointMethods?: EndpointMethod[];

  description?: string;

  created?: string;

  lastUpdated?: string;

  createdBy?: string;

  updatedBy?: string;

  deleted?: boolean;
}
